flocker/common/algebraic.py

```diff
@@ -0,0 +1,49 @@
+# -*- test-case-name: flocker.common.test.test_algebraic -*-
+# Copyright Hybrid Logic Ltd.  See LICENSE file for details.
+
+"""
+Invariants for algebraic data types.
+"""
```

Comment:
A lot of people will have to look up what "algebraic data type" means. It would really help to have an example in the docstring. Please add one.